Domain/SymfonyAclCache.php

```diff
@@ -0,0 +1,250 @@
+    {
+        $lookupKey = $this->getAliasKeyForIdentity($aclId);
+        if (!$this->cache->hasItem($lookupKey)) {
+            return;
+        }
+
+        $cacheItem = $this->cache->getItem($lookupKey);
+        if ($cacheItem->isHit()) {
+            $key = $cacheItem->get();
+            if ($this->cache->hasItem($key)) {
+                $this->cache->deleteItem($key);
+            }
+        }
+
+        $this->cache->deleteItem($lookupKey);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function evictFromCacheByIdentity(ObjectIdentityInterface $oid)
+    {
+        $key = $this->getDataKeyByIdentity($oid);
+        if (!$this->cache->hasItem($key)) {
+            return;
+        }
+
+        $this->cache->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFromCacheById($aclId)
+    {
+        $lookupKey = $this->getAliasKeyForIdentity($aclId);
+        if (!$this->cache->hasItem($lookupKey)) {
+            return;
+        }
+
+        $lookupKeyItem = $this->cache->getItem($lookupKey);
+        if (!$lookupKeyItem->isHit()) {
+            return;
+        }
+
+        $key = $lookupKeyItem->get();
+        if (!$this->cache->hasItem($key)) {
+            $this->cache->deleteItem($lookupKey);
+
+            return;
+        }
+
+        $keyItem = $this->cache->getItem($key);
+        if (!$keyItem->isHit()) {
+            $this->cache->deleteItem($lookupKey);
+
+            return;
+        }
+
+        return $this->unserializeAcl($keyItem->get());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFromCacheByIdentity(ObjectIdentityInterface $oid)
+    {
+        $key = $this->getDataKeyByIdentity($oid);
+        if (!$this->cache->hasItem($key)) {
+            return;
+        }
+
+        $cacheItem = $this->cache->getItem($key);
+        if (!$cacheItem->isHit()) {
+            return;
+        }
+
+        return $this->unserializeAcl($cacheItem->get());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function putInCache(AclInterface $acl)
+    {
+        if (null === $acl->getId()) {
+            throw new \InvalidArgumentException('Transient ACLs cannot be cached.');
+        }
+
+        if (null !== $parentAcl = $acl->getParentAcl()) {
+            $this->putInCache($parentAcl);
+        }
+
+        $key = $this->getDataKeyByIdentity($acl->getObjectIdentity());
+        $objectIdentityItem = $this->cache->getItem($key);
+        $objectIdentityItem->set(serialize($acl));
+
+        $this->cache->save($objectIdentityItem);
+
+        $aliasKey = $this->getAliasKeyForIdentity($acl->getId());
+        $aliasItem = $this->cache->getItem($aliasKey);
+        $aliasItem->set($key);
+
+        $this->cache->save($aliasItem);
+    }
+
+    /**
+     * Unserializes the ACL.
+     *
+     * @param string $serialized
+     *
+     * @return AclInterface
+     */
+    private function unserializeAcl($serialized)
```

Comment:
All the private methods look like C&P from the Doctrine cache class. Please move them into a trait that is consumed by both classes.

Reply:
Done